From 0eb1f83fd16cbda8223bc5047953acba5645869b Mon Sep 17 00:00:00 2001
From: Bart Oldeman <bart.oldeman@calculquebec.ca>
Date: Thu, 4 Mar 2021 00:33:20 +0000
Subject: [PATCH] Add HAVE_MPI_INFO_F2C to cmake config.h input file

Without this define parallel netCDF-Fortran did not work
correctly with Open MPI, for example
examples/F90/simple_xy_par_wr.f90
reported
*** An error occurred in MPI_Info_dup
*** reported by process [2543714305,2]
*** on communicator MPI_COMM_WORLD
*** MPI_ERR_INFO: invalid info object
*** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
***    and potentially your MPI job)

Fixes Unidata/netcdf-fortran#109
---
 config.h.cmake.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/config.h.cmake.in b/config.h.cmake.in
index 0b746e228..7ed01bf64 100644
--- a/config.h.cmake.in
+++ b/config.h.cmake.in
@@ -313,6 +313,9 @@ are set when opening a binary file on Windows. */
 /* Define to 1 if you have the `MPI_Comm_f2c' function. */
 #cmakedefine HAVE_MPI_COMM_F2C 1
 
+/* Define to 1 if you have the `MPI_Info_f2c' function. */
+#cmakedefine HAVE_MPI_INFO_F2C 1
+
 /* Define to 1 if you have the `mremap' function. */
 #cmakedefine HAVE_MREMAP 1
 
